packages/odata-common/test/common-entity.ts

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable max-classes-per-file */
+import {
+  Constructable,
+  EntityBase,
+  EntityBuilderType
+} from '../src/entity-base';
+import { CustomField } from '../src/selectable/custom-field';
+import { FieldBuilder } from '../src/selectable/field-builder';
+import { Field, FieldOptions } from '../src/selectable/field';
+import { EdmTypeField } from '../src/selectable/edm-type-field';
+import { ComplexTypeField } from '../src/selectable/complex-type-field';
+import { ConstructorOrField } from '../src/selectable/constructor-or-field';
+import { PropertyMetadata } from '../src/selectable/complex-type-namespace';
+import { AllFields } from '../src/selectable/all-fields';
+
+export interface CommonEntityComplexType {
+  stringProperty: string;
+  booleanProperty?: boolean;
+}
+export class CommonEntityComplexTypeField<
+  EntityT extends EntityBase
+> extends ComplexTypeField<EntityT, CommonEntityComplexType, true, true> {
+  _fieldBuilder: FieldBuilder<this> = new FieldBuilder(this);
+  stringProperty: EdmTypeField<EntityT, 'Edm.String', false, false> =
+    this._fieldBuilder.buildEdmTypeField('StringProperty', 'Edm.String', false);
+
+  booleanProperty: EdmTypeField<EntityT, 'Edm.Boolean', true, false> =
+    this._fieldBuilder.buildEdmTypeField(
+      'BooleanProperty',
+      'Edm.Boolean',
+      true
+    );
+  constructor(
+    fieldName: string,
+    fieldOf: ConstructorOrField<EntityT>,
+    fieldOptions?: FieldOptions<true, true>
+  ) {
+    super(fieldName, fieldOf, CommonEntityComplexType, fieldOptions);
+  }
+}
+export namespace CommonEntityComplexType {
+  export const _propertyMetadata: PropertyMetadata<CommonEntityComplexType>[] =
+    [
+      {
+        originalName: 'StringProperty',
+        name: 'stringProperty',
+        type: 'Edm.String',
+        isCollection: false
+      },
+      {
+        originalName: 'BooleanProperty',
+        name: 'booleanProperty',
+        type: 'Edm.Boolean',
+        isCollection: false
+      }
+    ];
+}
+
+// eslint-disable-next-line import/export
+export class CommonEntity extends EntityBase {
```

Comment:
Is it intended to have a different name "CommenEntity" v.s. "TestEntity"?

Reply:
I wanted to emphasise that it is not v2 or v4 entity but one for common.